DM-38893: Update lsst-texmf instructions for Overleaf

leeskelvin opened the pull request.

womullan commented on docs/overleaf.rst:
```diff
@@ -1,14 +1,46 @@
+
+.. code-block:: bash
+
+   rm texmf/tex/LSST-themes/default
```
I just set up another RTN now and i did not need to do anything with symlinks .. i also put the entire lsst-texmf in (removing .git files from root)  but that does not matter - I thought the themes were only for slide decks - are you doing a presentation ?

leeskelvin replied:
I've just ran `cp -av lsst-texmf/texmf texmf2` and committed this change to my locally checked out repo: https://github.com/leeskelvin/sitcomtn-069. When I then switch back to Overleaf and try to import this change, I get this pop-up error message:

> Your GitHub repository contains symbolic link files, which are not currently supported by Overleaf. Please remove these and try again.

leeskelvin replied:
PS - the Overleaf project is here: https://www.overleaf.com/9663157169wncdxfnrhmkk

womullan replied:
so i used the git from overleaf - cp -r the entire lsst-texmf  when I adde it somehow that link just becomes a file.  So you can simple rm that one sym link  no need to copy over it ..  not sure why overleaf git is so restrictive!

womullan reviewed (approved):
thanks for the updates !! Its great

leeskelvin commented on docs/overleaf.rst:
```diff
@@ -1,14 +1,46 @@
+.. code-block:: bash
+
+   cp -av lsst-texmf/texmf texmf
+
```
Good catch! I forgot to add notes on this (will update now). I did add `myacronyms.tex` manually, but did not add `meta.tex`. This file gets removed by `make clean`, which is why I'd missed it (and resorted to just commenting out the meta.tex line in the tex file).

leeskelvin commented on docs/overleaf.rst:
```diff
@@ -1,14 +1,46 @@
+
+.. code-block:: bash
+
+   git clone --recurse-submodules git@github.com/USERNAME/MYREPO.git
```
Yes, that's right - I wanted to avoid the situation whereby multiple perhaps temporary commits were being merged back into the main repo. As we may need to run `make` a number of times to update the `myacronyms.tex` file, there will be a potentially large number of commits to/from Overleaf that may cloud the git history.

leeskelvin replied:
I'd thought it would be cleaner to work entirely in a silod forked repo while the document is being developed, and then do a single merge at some future point when the document is ready. However, a fork is not necessary, and pushing directly to main has the advantage that the PDF will be built at the live link, so I'm happy to go that way if you think it better?

womullan replied:
i have been working with the git from overleaf  git clone https://git.overleaf.com/6442de95f32abd0a13d74afb   and i run make and commit and push to that. But i had not really though about going back i assumed i would put the files in a new branch of the repo and push it.  The import to overleaf makes a copy as far as i can see its not pushing changes back to your repo right ? so its about the same as your fork

leeskelvin replied:
Right, changes will eventually be pushed back to GitHub. Say I import my repo into Overleaf, make my edits in Overleaf, and then go to Menu -> Sync -> GitHub -> Push Overleaf changes to GitHub. This is how I get my Overleaf edits out of Overleaf and back into GitHub.

leeskelvin replied:
Oh, and there's another issue too that I'd forgotten about! When starting a new project on Overleaf, if I go to 'Import from GitHub', I don't currently see repos inside https://github.com/lsst-sitcom. I'm sure that's some kind of Overleaf/GitHub permissions issue that can be resolved, but in my case, I needed to fork the repo into my own userspace to make it visible to Overleaf.

leeskelvin commented:
Thanks for the initial review @womullan. I've added a new section regarding `meta.tex` and `acronyms.tex`. I've also rearranged the text somewhat to construct all files in advance of Overleaf (rather than making new files in Overleaf).

leeskelvin commented:
Thanks again @womullan! I've removed the directory removal text, and added some extra text regarding forking/not-forking. I've made it clear that forking the repo is optional, and not strictly required.

I've also added some extra info at the end on pushing edits out of Overleaf and pulling back to your local repo to update `acronyms.tex`. I hope that looks ok to you!
